src/utilities/filetypes/EpwFile.cpp

```diff
@@ -3948,13 +3960,8 @@ bool EpwFile::parse(std::istream& ifs, bool storeData) {
               m_minutesMatch = false;
             }
           }
-          boost::optional<EpwDataPoint> pt = EpwDataPoint::fromEpwStrings(year, month, day, hour, currentMinute, strings);
-          if (pt) {
-            m_data.push_back(pt.get());
-          } else {
-            LOG(Error, "Failed to parse line " << lineNumber << " of EPW file '" << m_path << "'");
-            return false;
-          }
+          epw_string epw_s = {lineNumber, year, month, day, hour, currentMinute, strings};
+          epw_strings.push_back(epw_s);
```

Comment:
I think you could save a copy by doing the construction in place. `push_back` will perform move and save a copy. Here it doesn't matter that much because the copy is presumably super cheap. Something like this...

```
epw_strings.push_back({lineNumber, year, month, day, hour, currentMinute, strings});
```

Reply:
` epw_strings.emplace_back(lineNumber, year, month, day, hour, currentMinute, std::move(strings));`

Reply:
A benchmark: https://github.com/jmarrec/CppBenchmarks/blob/main/bench_EPWStrings.cpp

![image](https://github.com/user-attachments/assets/58c78da7-fae3-4c1d-b1bc-3faacb288c73)

Reply:
The full range:

![image](https://github.com/user-attachments/assets/a79d514b-15c2-4cfc-80af-b3de9bdcb13c)